store/postgres/src/entities.rs

```diff
@@ -0,0 +1,993 @@
+                ))
+                .do_update()
+                .set((
+                    public::entities::data.eq(data),
+                    public::entities::event_source.eq(&event_source.to_string()),
+                ))
+                .execute(conn)?),
+            Table::Split(entities) => {
+                let query = format!(
+                    "INSERT into {}.entities(entity, id, data, event_source)
+                                VALUES($1, $2, $3, $4)
+                                ON CONFLICT(entity, id)
+                                DO UPDATE SET data = $3, event_source = $4",
+                    entities.schema
+                );
+                let query = diesel::sql_query(query)
+                    .bind::<Text, _>(&key.entity_type)
+                    .bind::<Text, _>(&key.entity_id)
+                    .bind::<Jsonb, _>(data)
+                    .bind::<Text, _>(event_source.to_string());
+                Ok(query.execute(conn)?)
+            }
+        }
+    }
+
+    fn update(
+        &self,
+        conn: &PgConnection,
+        key: &EntityKey,
+        data: &serde_json::Value,
+        guard: Option<EntityFilter>,
+        event_source: EventSource,
+    ) -> Result<usize, StoreError> {
+        match self {
+            Table::Public(subgraph) => {
+                let target = public::entities::table
+                    .filter(public::entities::subgraph.eq(subgraph.to_string()))
+                    .filter(public::entities::entity.eq(&key.entity_type))
+                    .filter(public::entities::id.eq(&key.entity_id));
+
+                let query = diesel::update(target).set((
+                    public::entities::data.eq(public::entities::data.merge(data)),
+                    public::entities::event_source.eq(event_source.to_string()),
+                ));
+
+                match guard {
+                    Some(filter) => {
+                        let filter = build_filter(filter).map_err(|e| {
+                            TransactionAbortError::Other(format!(
+                                "invalid filter '{}' for value '{}'",
+                                e.filter, e.value
+                            ))
+                        })?;
+                        Ok(query.filter(filter).execute(conn)?)
+                    }
+                    None => Ok(query.execute(conn)?),
+                }
+            }
+            Table::Split(entities) => entities.update(conn, key, data, guard, event_source),
+        }
+    }
+
+    fn delete(
+        &self,
+        conn: &PgConnection,
+        key: &EntityKey,
+        event_source: EventSource,
+    ) -> Result<usize, StoreError> {
+        diesel::select(set_config(
+            "vars.current_event_source",
+            event_source.to_string(),
+            true,
+        ))
+        .execute(conn)
+        .map_err(|e| format_err!("Failed to save event source for remove operation: {}", e))
```

Comment:
Could we call this 'set' instead of 'save'.

Reply:
Yep, done.